lib/core/include/scipp/core/element/event_operations.h

```diff
@@ -118,13 +118,11 @@ constexpr auto map_and_mul = overloaded{
     element::arg_list<
         map_and_mul_detail::args<double, double, double, double>,
         map_and_mul_detail::args<double, double, double, float>,
-        map_and_mul_detail::args<float, double, double, double>,
         map_and_mul_detail::args<float, double, double, float>,
         map_and_mul_detail::args<double, float, float, double>,
         map_and_mul_detail::args<double, time_point, time_point, double>,
         map_and_mul_detail::args<double, time_point, time_point, float>,
-        map_and_mul_detail::args<float, time_point, time_point, double>,
-        map_and_mul_detail::args<float, time_point, time_point, float>>,
+        map_and_mul_detail::args<float, time_point, time_point, double>>,
```

Comment:
Why have some combinations been removed?

Reply:
They dealt with type promotions which are now handled in Python.

Reply:
What about the inplace variants (which you did not change)?

Reply:
In place operations cannot change the dtype, can they?

Reply:
No they cannot, but how is that related to the code and my question?

Reply:
Which in-place variants do you actually mean? This is only used by `dataset::buckets::scale` which is always in-place.

Reply:
For example `__imul__` in `bins.py`, right under the functions you changed in this PR.

Reply:
And I actually removed the wrong one for the time_point overload.